src/Integration/Persistence/SolutionBindingSerializer.cs

```diff
@@ -76,7 +79,7 @@ private void AddSolutionItemFile(string configFile)
             var projectSystemHelper = this.serviceProvider.GetService<IProjectSystemHelper>();
             projectSystemHelper.AssertLocalServiceIsNotNull();
 
-            Project solutionItemsProject = projectSystemHelper.GetSolutionFolderProject(Constants.SonarQubeManagedFolderName, true);
+            Project solutionItemsProject = projectSystemHelper.GetSolutionFolderProject(Constants.LegacySonarQubeManagedFolderName, true);
```

Comment:
There are separate serializer classes for legacy and connected modes, so the binding mode can be hard-coded.